x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/SearchableSnapshotDirectoryTests.java

```diff
@@ -0,0 +1,361 @@
+                    assertThat(searchableReader.leaves(), hasSize(reader.leaves().size()));
+                    assertThat(searchableReader.maxDoc(), equalTo(reader.maxDoc()));
+                    assertThat(searchableReader.getVersion(), equalTo(reader.getVersion()));
+                    assertThat(searchableReader.getIndexCommit().getGeneration(), equalTo(reader.getIndexCommit().getGeneration()));
+
+                    String field = randomFrom("id", "text");
+                    Terms terms = reader.leaves().get(0).reader().terms(field);
+                    Terms searchableTerms = searchableReader.leaves().get(0).reader().terms(field);
+                    assertThat(searchableTerms.size(), equalTo(terms.size()));
+                    assertThat(searchableTerms.getDocCount(), equalTo(terms.getDocCount()));
+                    assertThat(searchableTerms.getMin(), equalTo(terms.getMin()));
+                    assertThat(searchableTerms.getMax(), equalTo(terms.getMax()));
+                }
+            }
+        });
+    }
+
+    public void testReadByte() throws Exception {
+        testIndexInputs((indexInput, searchableIndexInput) -> {
+            try {
+                for (int i = 0; i < 10; i++) {
+                    if (randomBoolean()) {
+                        long position = randomLongBetween(0L, indexInput.length());
+                        indexInput.seek(position);
+                        searchableIndexInput.seek(position);
+                    }
+                    assertThat("File pointers values should be the same before reading a byte",
+                        searchableIndexInput.getFilePointer(), equalTo(indexInput.getFilePointer()));
+
+                    if (indexInput.getFilePointer() < indexInput.length()) {
+                        assertThat(searchableIndexInput.readByte(), equalTo(indexInput.readByte()));
+                    } else {
+                        expectThrows(EOFException.class, searchableIndexInput::readByte);
+                    }
+                    assertThat("File pointers values should be the same after reading a byte",
+                        searchableIndexInput.getFilePointer(), equalTo(indexInput.getFilePointer()));
+                }
+            } catch (IOException e) {
+                throw new AssertionError(e);
+            }
+        });
+    }
+
+    public void testReadBytes() throws Exception {
+        final byte[] buffer = new byte[8192];
+        final byte[] searchableBuffer = new byte[buffer.length];
+
+        testIndexInputs((indexInput, searchableIndexInput) -> {
+            try {
+                if (randomBoolean()) {
+                    long position = randomLongBetween(0L, indexInput.length());
+                    indexInput.seek(position);
+                    searchableIndexInput.seek(position);
+                }
+                assertThat("File pointers values should be the same before reading a byte",
```

Comment:
Could we include `searchableIndexInput.toString()` and `indexInput.toString()` in the assertion message (and elsewhere in this test too).